runners/spark/src/main/java/org/apache/beam/runners/spark/translation/streaming/SparkRunnerStreamingContextFactory.java

```diff
@@ -95,6 +96,25 @@ public JavaStreamingContext create() {
     return jssc;
   }
 
+  private void checkpoint(JavaStreamingContext jssc) {
+    Path rootCheckpointPath = checkpointDir.getRootCheckpointDir();
+    Path sparkCheckpointPath = checkpointDir.getSparkCheckpointDir();
+
+    try {
+      FileSystem fileSystem = rootCheckpointPath.getFileSystem(jssc.sc().hadoopConfiguration());
+      if (!fileSystem.exists(rootCheckpointPath)) {
+        fileSystem.mkdirs(rootCheckpointPath);
+      }
+      if (!fileSystem.exists(sparkCheckpointPath)) {
+        fileSystem.mkdirs(sparkCheckpointPath);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Failure while writing to checkpoint dir", e);
```

Comment:
This isn't necessarily a failure to create a checkpoint dir, but could be from `getFileSystem()` as well as either of the `mkdir()`s.

I suppose I can surround each of the 3 calls with separate `try` `catch` clauses, however this could make the code here really messy.

The exceptions thrown by these methods seem to be sufficiently verbose, and are added as the cause exception to the thrown `RuntimeException` (e).

Is this sufficient?